packages/dev/core/src/Maths/math.vector.ts

```diff
@@ -2,33 +2,205 @@
 import { Scalar } from "./math.scalar";
 import { Epsilon } from "./math.constants";
 import type { Viewport } from "./math.viewport";
-import type { DeepImmutable, Nullable, FloatArray, float } from "../types";
+import type { DeepImmutable, Nullable, FloatArray, float, Constructor } from "../types";
 import { ArrayTools } from "../Misc/arrayTools";
 import type { IPlaneLike } from "./math.like";
 import { RegisterClass } from "../Misc/typeStore";
 import type { Plane } from "./math.plane";
 import { PerformanceConfigurator } from "../Engines/performanceConfigurator";
 import { EngineStore } from "../Engines/engineStore";
 import type { TransformNode } from "../Meshes/transformNode";
-
-export type Vector2Constructor<T extends Vector2> = new (...args: ConstructorParameters<typeof Vector2>) => T;
-export type Vector3Constructor<T extends Vector3> = new (...args: ConstructorParameters<typeof Vector3>) => T;
-export type Vector4Constructor<T extends Vector4> = new (...args: ConstructorParameters<typeof Vector4>) => T;
-export type QuaternionConstructor<T extends Quaternion> = new (...args: ConstructorParameters<typeof Quaternion>) => T;
-export type MatrixConstructor<T extends Matrix> = new () => T;
+import { Tensor } from "./tensor";
 
 // eslint-disable-next-line @typescript-eslint/naming-convention
 const _ExtractAsInt = (value: number) => {
     return parseInt(value.toString().replace(/\W/g, ""));
 };
 
+/**
+ * Reprents a vector of any dimension
+ */
+export declare abstract class Vector<N extends number[] = number[]> extends Tensor<N> {

```

Comment:
Since this is used with `implements`, would it be better to be defined as an interface?

Reply:
It's defined as a declare class so the static members exist (which is very important when using `typeof Class`.